From e0823eb50c1f45b8ed8aa39d15bed29b2e239742 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Tue, 10 Oct 2017 16:31:38 +0300
Subject: [PATCH 2030/2241] fix: dma: mv_xor_v2: unregister xor clock when
 remove the xor driver

This patch add call function while removing the xor driver to unregister
the xor clock.

Change-Id: Idd2ac19c1dff60a8f9cf731b723a649c7f3f7f2e
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/45097
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 drivers/dma/mv_xor_v2.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/dma/mv_xor_v2.c b/drivers/dma/mv_xor_v2.c
index ee57a5c..e6fcaff 100644
--- a/drivers/dma/mv_xor_v2.c
+++ b/drivers/dma/mv_xor_v2.c
@@ -1063,6 +1063,8 @@ static int mv_xor_v2_remove(struct platform_device *pdev)
 
 	platform_msi_domain_free_irqs(&pdev->dev);
 
+	clk_disable_unprepare(xor_dev->clk);
+
 	return 0;
 }
 
-- 
2.7.4

